server/src/main/java/io/druid/query/extraction/LookupExtractionModule.java

```diff
@@ -0,0 +1,166 @@
+import com.google.common.base.Function;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import io.druid.guice.Jerseys;
+import io.druid.guice.LazySingleton;
+import io.druid.guice.LifecycleModule;
+import io.druid.guice.annotations.Json;
+import io.druid.guice.annotations.Self;
+import io.druid.guice.annotations.Smile;
+import io.druid.initialization.DruidModule;
+import io.druid.server.DruidNode;
+import io.druid.server.listener.announcer.ListenerResourceAnnouncer;
+import io.druid.server.listener.announcer.ListeningAnnouncerConfig;
+import io.druid.server.listener.resource.AbstractListenerHandler;
+import io.druid.server.listener.resource.ListenerResource;
+import io.druid.server.namespace.cache.LookupCoordinatorManager;
+import org.apache.curator.framework.CuratorFramework;
+
+import javax.ws.rs.Path;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Include when a service can listen to lookups
+ */
+public class LookupExtractionModule implements DruidModule
+{
+  @Override
+  public List<? extends Module> getJacksonModules()
+  {
+    return ImmutableList.of(new SimpleModule("LookupExtracitonModule-" + UUID.randomUUID().toString()));
+  }
+
+  @Override
+  public void configure(Binder binder)
+  {
+    binder.bind(LookupReferencesManager.class).in(LazySingleton.class);
+    binder.bind(LookupListeningResource.class).in(LazySingleton.class);
+    binder.bind(LookupResourceListenerAnnouncer.class).in(LazySingleton.class);
+    LifecycleModule.register(binder, LookupReferencesManager.class);
+    LifecycleModule.register(binder, LookupListeningResource.class);
+    LifecycleModule.register(binder, LookupResourceListenerAnnouncer.class);
+    Jerseys.addResource(binder, LookupListeningResource.class);
+  }
+}
+
+
+@Path(ListenerResource.BASE_PATH + "/" + LookupCoordinatorManager.LOOKUP_LISTEN_ANNOUNCE_KEY)
+class LookupListeningResource extends ListenerResource
+{
+  @Inject
+  public LookupListeningResource(
+      final @Json ObjectMapper jsonMapper,
+      final @Smile ObjectMapper smileMapper,
+      final LookupReferencesManager manager
+  )
+  {
+    super(
+        jsonMapper,
+        smileMapper,
+        new AbstractListenerHandler<LookupHolder>(new TypeReference<LookupHolder>()
+        {
+        })
+        {
+          @Override
+          public Map<String, Object> post(final List<LookupHolder> inputObject) throws Exception
+          {
+            final Map<String, LookupExtractorFactory> lookups = new HashMap<>();
+            for (LookupHolder holder : inputObject) {
+              final String name = holder.name;
+              final LookupExtractorFactory factory = holder.factory;
+              lookups.put(name, factory);
+            }
+            manager.put(lookups);
```

Comment:
that is true. In such a case what would be the desired behavior?

My initial thought was the coordinating unit would have to reconcile the difference in expected vs actual state on its own.